mcp/runtime/src/main/java/io/quarkiverse/langchain4j/mcp/runtime/config/McpRegistryClientBuildTimeConfig.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkiverse.langchain4j.mcp.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocIgnore;
+import io.quarkus.runtime.annotations.ConfigGroup;
+
+/**
+ * Empty right now. Necessary to fix the list of registry client configuration keys at build time.
+ */
+@ConfigGroup
+public interface McpRegistryClientBuildTimeConfig {
```

Comment:
@geoand  maybe you can advise how to get rid of this class. But I had to add "some" build time config to be able to access a map of the client configuration keys during build (to register them in CDI). I don't know if there's a better solution than an empty build-time config class.

Reply:
Ah! I don't know TBH, I've not encountered this. @radcortez what is the best way to handle this situation?

Reply:
Roberto is off today, so depending on how soon you want to do the final release, we may want to merge this with this hack

Reply:
No, the users don't have to set anything. I had to include an actual config property because otherwise the `Map<String, McpRegistryClientBuildTimeConfig>` appears empty in the processor, but I made that property optional.

Reply:
But @jmartisk only wants the "structure", not the actual values, right @jmartisk ?

Reply:
I would like the values to be runtime config, but I need the list of keys at build time - to register them in CDI (I do know that this means you can't add new keys at runtime, but I guess that will simply be a limitation)

> Is it required to have `@McpRegistryClientName("CLIENT1")`, where I know that the keys that show up in the configuration must match with the annotation?

that's why I want the keys available at build time, to be able to register the beans with the `@McpRegistryClientName("CLIENT1")` qualifier

Reply:
And what about getting the keys from the annotations instead of Config?

Reply:
That would become a problem if somebody accesses the client only programmatically (and so there's no injection point with an annotation). And if somebody accidentally makes a typo on one side (the name in the annotation and the configuration key won't be the same), then we won't be able to detect it (and they will inject a badly configured client).

Reply:
Anyway, we are discussing two different problems here:

- The original one is how to get rid of that subgroup. If we know all the keys are available, you can use a simple `Map<String, Map<String, String>>`. Why a double Map? Because Config, will take the path and adjust the rest of the segment to the type being mapped, so something like `quarkus.langchain4j.mcp.registry-client.CLIENT1.base`, will store `CLIENT1.base` as a key in a single `Map<String, String>`. With `Map<String, Map<String, String>>`, it will use the next segment of the path for the first `Map` key and adjust the remaining segments to the second `Map`. Since you only want the keys, this would be a way to do it.

- As for the second issue, I'll let you guys decide what is acceptable in this situation.

Reply:
Now there's a totally weird error when generating documentation... ugh

Reply:
:(

Reply:
Ah, that was caused by a missing javadoc comment (though the error message was completely useless)